File: sol/solution/0001-0100/0001. Two Sum/Solution.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var d = [Int: Int]()
        for (i, x) in nums.enumerated() {
            if let j = d[target - x] {
                return [j, i]
            }
            d[x] = i
        }
        return []
    }
}

File: sol/solution/0001-0100/0001. Two Sum/Solution.scala
import scala.collection.mutable
object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        val d = mutable.Map[Int, Int]()
        var ans: Array[Int] = Array()
        for (i <- nums.indices if ans.isEmpty) { 
            if (d.contains(target - nums(i))) {
                ans = Array(d(target - nums(i)), i)
            } else {
                d(nums(i)) = i
            }
        }
        ans
    }
}